Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -248,11 +248,18 @@ public class PBXProjGenerator {
                 addObject(targetDependency)
                 dependencies.append(targetDependency.reference)
 
-                // don't bother linking a target dependency
-                // let dependencyBuildFile = targetBuildFileReferences[dependencyTargetName]!
-                // targetFrameworkBuildFiles.append(dependencyBuildFile)
+                if dependencyTarget.type.isLibrary || dependencyTarget.type.isFramework {
+                    let dependencyBuildFile = targetBuildFiles[dependencyTargetName]!
+                    let buildFile = PBXBuildFile(reference: generateUUID(PBXBuildFile.self, dependencyBuildFile.reference + target.name), fileRef: dependencyBuildFile.fileRef)
+                    addObject(buildFile)
+                    targetFrameworkBuildFiles.append(buildFile.reference)
+                }
 
                 if embed {
+                    if dependencyTarget.type.isLibrary {
```

Comment:
Is a library _never_ embeded, or are there special cases where they are? 
If it can be sometimes, ca we move this logic up into where `let embed` is defined, which gives the embedding a default, and then users can override this.

If it's definitely never, we can also move this if into the if above `if embed && !dependencyTarget.type.isLibrary`

Reply:
They are never embedded. Since that would just mean that you're literally copying the `.a` file into your `.app`, which would never be used at runtime since static libraries are linked into your final binary instead. I've updated the condition!